src/commands/utils/graph.ts

```diff
@@ -0,0 +1,230 @@
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+export async function getServicePrincipalsForApp(
+    graphClient: Client,
+    appId: string,
+): Promise<Errorable<ServicePrincipal[]>> {
+    try {
+        const spSearchResults: GraphListResult<ServicePrincipal> = await graphClient
+            .api("/servicePrincipals")
+            .filter(`appId eq '${appId}'`)
+            .get();
+
+        return { succeeded: true, result: spSearchResults.value };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+export async function getFederatedIdentityCredentials(
+    graphClient: Client,
+    applicationId: string,
+): Promise<Errorable<FederatedIdentityCredential[]>> {
+    try {
+        const identityResults: GraphListResult<FederatedIdentityCredential> = await graphClient
+            .api(`/applications/${applicationId}/federatedIdentityCredentials`)
+            .get();
+
+        return { succeeded: true, result: identityResults.value };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+export async function createFederatedIdentityCredential(
+    graphClient: Client,
+    applicationId: string,
+    subject: string,
+    name: string,
+    description: string,
+): Promise<Errorable<FederatedIdentityCredential>> {
+    const newCred: FederatedIdentityCredentialParams = {
+        name,
+        subject,
+        issuer: federatedIdentityCredentialIssuer,
+        description,
+        audiences: [federatedIdentityCredentialAudience],
+    };
+
+    try {
+        const cred: FederatedIdentityCredential = await graphClient
+            .api(`/applications/${applicationId}/federatedIdentityCredentials`)
+            .post(newCred);
+
+        return { succeeded: true, result: cred };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+} ////////TODO: catch fail logic if the same cred already exists
+
+export async function createGitHubActionFederatedIdentityCredential(
+    graphClient: Client,
+    applicationId: string,
+    organization: string,
+    repository: string,
+    branch: string,
+): Promise<Errorable<FederatedIdentityCredential>> {
+    const subject = `repo:${organization}/${repository}:ref:refs/heads/${branch}`;
+    return createFederatedIdentityCredential(graphClient, applicationId, subject, "gitHub_actions", "");
+}
+
+export async function deleteFederatedIdentityCredential(
+    graphClient: Client,
+    applicationId: string,
+    credId: string,
+): Promise<Errorable<void>> {
+    try {
+        await graphClient.api(`/applications/${applicationId}/federatedIdentityCredentials/${credId}`).delete();
+        return { succeeded: true, result: undefined };
+    } catch (e) {
+        return { succeeded: false, error: getErrorMessage(e) };
+    }
+}
+
+export function findFederatedIdentityCredential(
```

Comment:
💡 Where is this function used?

Reply:
Currently not used in playground test file but can be used in the future as a check step to not overwrite or recreate the same credential.